executable/detect.go

```diff
@@ -65,9 +65,15 @@ func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error
 		return libcnb.DetectResult{}, fmt.Errorf("unable to read manifest in %s\n%w", context.Application.Path, err)
 	}
 
+	jarGlob, _ := cr.Resolve("BP_EXECUTABLE_JAR_LOCATION")
```

Comment:
Could we move those 2 lines into the `else`? Searching for a `jar` is only relevant if no "Main-Class" is found yet.

So something like

```
if m.Get("Main-Class") {
//case 1
} else {
  //search jar
 if found {
  //case 2
  }
}

Reply:
The same logic is already contained in https://github.com/paketo-buildpacks/executable-jar/blob/bba2e76c45e18e40d76181c40ba3682a91e9d12f/executable/executable_jar.go#L40

The detection logic can be minimised down to:

```golang
jarGlob, _ := cr.Resolve("BP_EXECUTABLE_JAR_LOCATION")
execJar, err := LoadExecutableJAR(context.Application.Path, jarGlob)
if err != nil {
	return libcnb.DetectResult{}, fmt.Errorf("unable to load executable JAR\n%w", err)
}

if !reflect.DeepEqual(execJar, ExecutableJAR{}) {
	d.Logger.Info("PASSED: 'Main-Class' manifest attribute found")
	result.Plans[0].Provides = append(result.Plans[0].Provides, libcnb.BuildPlanProvide{Name: PlanEntryJVMApplicationPackage})
}
```

Reply:
Thanks, updated :)

Reply:
the idea was to replace everything with the suggested snippet, starting from the line 64:

https://github.com/paketo-buildpacks/executable-jar/blob/4178165057eb0603fa6e68e35e43ac12cbc45043/executable/detect.go#L64-L83

Reply:
@pbusko Got you, thanks; updated.